src/main/java/fi/otavanopisto/kuntaapi/server/integrations/ptv/translation/KuntaApiPtvTranslator.java

```diff
@@ -259,20 +311,177 @@
       .filter(Objects::nonNull)
       .collect(Collectors.toList());
   }
+
+  /**
+   * Translates ontolygy items into uris
+   * 
+   * @param items items
+   * @return uris
+   */
+  public List<String> translateOntologyItems(List<OntologyItem> fintoItems) {
+    if (fintoItems == null) {
+      return Collections.emptyList();
+    }
+
+    return fintoItems.stream()
+      .map(OntologyItem::getUri)
+      .filter(Objects::nonNull)
+      .collect(Collectors.toList());
+  }
+
+  /**
+   * Translates Kunta API laws into PTV laws
+   * 
+   * @param laws laws
+   * @return
+   */
+  public List<V4VmOpenApiLaw> translateLaws(List<Law> laws) {
+    if (laws == null) {
+      return Collections.emptyList();
+    }
+
+    return laws.stream()
+      .map(this::translateLaw)
+      .filter(Objects::nonNull)
+      .collect(Collectors.toList());
+  }
+
+  /**
+   * Translates Kunta API vouchers into PTV vouchers
+   * 
+   * @param vouchers Kunta API vouchers
+   * @return PTV vouchers
+   */
+  public List<VmOpenApiServiceVoucher> translateVouchers(List<ServiceVoucher> vouchers) {
+    if (vouchers == null) {
+      return Collections.emptyList();
+    }
+    
+    List<VmOpenApiServiceVoucher> result = new ArrayList<>(vouchers.size());
+    
+    for (int i = 0; i < vouchers.size(); i++) {
+      VmOpenApiServiceVoucher voucher = translateVoucher(vouchers.get(i), i);
+      if (voucher != null) {
+        result.add(voucher);
+      }
+    }
+    
+    return result;
+  }
+  
+  /**
+   * Translates Kunta API service organizations into PTV Service Producers
+   * 
+   * @param organizations Kunta API service organizations
+   * @return PTV Service Producers
+   */
+  public List<VmOpenApiServiceProducerIn> translateServiceProducers(List<ServiceOrganization> organizations) {
+    if (organizations == null) {
+      return Collections.emptyList();
+    }
+    
+    List<ServiceOrganization> serviceProducers = organizations.stream()
+      .filter((serviceOrganization) -> !" ".equals(serviceOrganization.getRoleType()))
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "serviceOrganization" parameter [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1611)